### PR TITLE
Form feature

### PR DESCRIPTION
**Pull request by gango-anan**
# The form-feature involved:
- Creating the form using plain html
- Creating the same form using the Rails #form_at and #form_for Helpers.
- Add a edit feature plus its view.

### CODE REVIEW

**Review by NtwaliHeritier — CHANGES_REQUESTED** [review]
### STATUS: CHANGES REQUESTED🛑
Hello team,
This is some great work, I liked the way you covered almost all parts of the project.
However there is a small minor change to make to make the project outstanding.

[REQUIRED] : On the HTML form, on action, put "users_path" instead of create and test the output.
Make those changes and ask for another review.
Happy coding 🧨

**Comment by gango-anan** [comment]
Great review, changes made and submitted, thanks.

File: app/views/users/new.html.erb [hunk 1]
```diff
@@ -1,2 +1,44 @@
-<h1>Users#new</h1>
-<p>Find me in app/views/users/new.html.erb</p>
+<h1>New User</h1>
+
+<!-- HTML Form -->
+<%
+=begin%>
+ <form action="create" method="post" accept-charset="UTF-8">
```
Comment (enaburekhan) [inline app/views/users/new.html.erb]: Kindly change "create" to "users_path"

**Review by enaburekhan — CHANGES_REQUESTED** [review]
# Status: Request changes :full_moon:
Good job so far:clap::clap:. However, you didn't implement all the required changes requested by the previous Tse. 
**Mandatory requirements**
Kindly use the `users_path` helper for action in the commented out HTML form as already highlighted in the codebase below.

[Optional] 
- form view to displays a list of the error messages (with #errors and full_messages usage) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_extra_shared_error.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
- validate the email format by using either a regular expression or a built-in URI method.

Kindly make the required changes and request another review when you are done.

You can reach me on slack @ Eric Enaburekhan if you have any questions.

Happy coding :blush:
Eric Enaburekhan

**Comment by descholar-ceo** [comment]
# This Pull Request is under review by [descholar-ceo](https://github.com/descholar-ceo) :eyes: :eyes: 
## Please wait a bit ...

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -1,2 +1,46 @@
-<h1>Users#new</h1>
-<p>Find me in app/views/users/new.html.erb</p>
+<h1>New User</h1>
+
+<!-- HTML Form -->
+
+<%
+=begin%>
+ <form action="create" method="post" accept-charset="UTF-8">
```
Comment (descholar-ceo) [inline app/views/users/new.html.erb]: [Required] Replace the create with `users_path` helper method for rails

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
# Status: Changes Required :vertical_traffic_light: :warning: 
## Hello @gango-anan  :handshake: :handshake: :handshake: 

:+1: This is good to make it this far, your project is really nice, congratulations on the good work you have done so far.
:+1: I liked that you have tried to implemented the requested changes from the previous review, keep it up, and your project is really almost done.
#### :warning: However we need to make few changes to mark this project approved, don't feel discouraged because of being asked some changes more and more, that's beacuse this is a learning process, and we are here to help you to become an out-standing developer. See the requested change below :point_down: :point_down: :point_down: 
### Note: All of requested changes which are marked as [Required] are mandatory to be implemented in order to get this project approved, others which are marked as [Optional] are nice to have in your project to make it outstanding

# [Required Changes] :red_circle: 
- [ ] Fix the MIT license link from the readme file => currently the one you have there is broken
- [ ] Fix some typo errors from the readme file, you can use [grammarly](https://app.grammarly.com/) to spot those errors
- [ ] Read the inline comment that I highlighted about the replacement of `create` with `users_path` helper method
___
_If anything seems to be unclear to you, reachout and ask anything by commenting on this Pull Request and mention me as [descholar-ceo](https://github.com/descholar-ceo), I will receive a notification and will respond._

[GitHub](https://github.com/descholar-ceo) | [LinkedIn](https://linkedin.com/in/mugirase-emmanuel) | [Twitter](https://twitter.com/descholar3)

_Reviewed by [Mugirase Emmanuel (descholar)](https://github.com/descholar-ceo)_

File: app/views/users/new.html.erb (hunk 1 above)
Comment (gango-anan) [reply app/views/users/new.html.erb]: Made that change but forgot to push the changes before requesting for the review, thanks for the observation. 
The requested changes have been made and another review requested.

**Review by cristianCeamatu — APPROVED** [review]
### 🥇 APPROVED 🥇 Congratulations and good luck with the next projects!

If you ever need any help or notice that you are wasting too many hours on a bug don't hesitate to contact me on Slack.

Best regards,
Cristian Ceamatu
Slack: Cristian Viorel Ceamatu

**Pull request merged**